Source/Common/singleton.cpp

```diff
@@ -65,8 +66,15 @@ void init_http_singleton()
             &httpSingleton,
             newSingleton
         );
```

Comment:
Just realised this code does not allocate memory the right way, we should pass an allocator to make shared...

Reply:
Will do